### PR TITLE
BUG: Duplicated returns boolean dataframe

### LINKED ISSUE #25184: Filtering by exclusion of duplicate rows does not preserve column list for an empty dataframe
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

x_df = pd.DataFrame(columns=['a', 'b'])
series = x_df.duplicated(subset=['a'])

list(x_df[~series])

# Expected output on Pandas 0.23.4: ['a', 'b']
# But, Pandas 0.24.1 returns: []
```
#### Problem description

We have been using this approach to remove duplicate rows on a dataframe, where rows are compared by one column only. Everything worked perfectly until we found out that, if the original dataframe is empty, in the result dataframe column list is lost after Pandas upgrade to latest version.

#### Expected Output

We would expect the column list to be preserved.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.15.final.0
python-bits: 64
OS: Linux
OS-release: 4.15.0-44-generic
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: C.UTF-8
LOCALE: None.None

pandas: 0.24.1
pytest: None
pip: 18.0
setuptools: 40.0.0
Cython: 0.22
numpy: 1.12.1
scipy: None
pyarrow: None
xarray: None
IPython: 4.2.0
sphinx: 1.4.4
patsy: None
dateutil: 2.5.0
pytz: 2017.2
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: 2.5.14
xlrd: 1.2.0
xlwt: 1.3.0
xlsxwriter: None
lxml.etree: 3.7.3
bs4: 4.5.3
html5lib: 1.
</details>

### PR DESCRIPTION
**Pull request by gnilrets**
- [x] closes #25184
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.25.0.rst [hunk 1]
```diff
@@ -68,6 +68,8 @@ Performance Improvements
 Bug Fixes
 ~~~~~~~~~
 
+- Bug in using duplicated with empty dataframes (:issue:`25184`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: say more like

``:meth:DataFrame.duplicate()`` on empty dataframe was not return a boolean dtyped Series

File: pandas/tests/frame/test_duplicates.py [hunk 2]
```diff
@@ -182,6 +182,12 @@ def test_drop_duplicates():
         assert df.duplicated(keep=keep).sum() == 0
 
 
+def test_duplicated_on_empty_frame_gives_back_frame():
```
Comment (jreback) [inline pandas/tests/frame/test_duplicates.py]: can you add the issue number as a comment here, rename the test to

test_duplicated_on_empty_frame

File: pandas/tests/frame/test_duplicates.py [hunk 3]
```diff
@@ -182,6 +182,12 @@ def test_drop_duplicates():
...
+    df = DataFrame(columns=['a', 'b'])
+    dupes = df.duplicated('a')
```
Comment (jreback) [inline pandas/tests/frame/test_duplicates.py]: result = 
expected = 
tm.assert_frame_equal(result, expected)

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.25.0.rst (hunk 1 above)
Comment (jreback) [reply doc/source/whatsnew/v0.25.0.rst]: this can go in 0.24.2

**Comment by gnilrets** [comment]
@jreback - Requested changes addressed.

I wasn't sure if this PR should go against master or the v0.24.x branch?

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @gnilrets all PRs go against master; *some* are backported.

**Pull request merged**